[Automation] javac compile fails for io.opentelemetry:opentelemetry-exporter-common:1.28.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.28.0" ./gradlew clean compileTestJava -Pcoordinates="io.opentelemetry:opentelemetry-exporter-common:1.19.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25027281349/job/73301240272
Last 300 lines of the log:
```console
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         2 tests found           ]
[         0 tests skipped         ]
[         2 tests started         ]
[         0 tests aborted         ]
[         2 tests successful      ]
[         0 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 12s
15 actionable tasks: 7 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
-------
Test for io.opentelemetry:opentelemetry-exporter-common:1.19.0 passed.
====================

> Task :test

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 14s
11 actionable tasks: 3 executed, 8 up-to-date
PASSED:1.27.0
========================================================================================
 Testing io.opentelemetry:opentelemetry-exporter-common:1.19.0:1.28.0
========================================================================================
========================================================================================
 io.opentelemetry:opentelemetry-exporter-common:1.19.0:1.28.0 stage native-image run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 810ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest FAILED
====================
Testing library: io.opentelemetry:opentelemetry-exporter-common:1.19.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                               ^
  symbol:   class RetryPolicy
  location: package io.opentelemetry.exporter.internal.retry
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                        () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                         ^
  symbol:   variable RetryPolicy
  location: class RetryUtilTest
2 errors

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

FAILURE: 
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
  import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                                 ^
    symbol:   class RetryPolicy
    location: package io.opentelemetry.exporter.internal.retry
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                          () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                           ^
    symbol:   variable RetryPolicy
    location: class RetryUtilTest
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.opentelemetry:opentelemetry-exporter-common:1.19.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 1.28.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.28.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.28.0" ./gradlew clean test -Pcoordinates="io.opentelemetry:opentelemetry-exporter-common:1.19.0"]
========================================================================================
 io.opentelemetry:opentelemetry-exporter-common:1.19.0:1.28.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 778ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                               ^
  symbol:   class RetryPolicy
  location: package io.opentelemetry.exporter.internal.retry
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                        () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                         ^
  symbol:   variable RetryPolicy
  location: class RetryUtilTest
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
  import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                                 ^
    symbol:   class RetryPolicy
    location: package io.opentelemetry.exporter.internal.retry
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                          () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                           ^
    symbol:   variable RetryPolicy
    location: class RetryUtilTest
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

BUILD FAILED in 856ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'javac compile' for 1.28.0 failed with exit code 1.
FAILED[javac compile][1.28.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.28.0" ./gradlew clean compileTestJava -Pcoordinates="io.opentelemetry:opentelemetry-exporter-common:1.19.0"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.28.0" ./gradlew clean compileTestJava -Pcoordinates="io.opentelemetry:opentelemetry-exporter-common:1.19.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25027281349/job/73301240360
Last 300 lines of the log:
```console
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         2 tests found           ]
[         0 tests skipped         ]
[         2 tests started         ]
[         0 tests aborted         ]
[         2 tests successful      ]
[         0 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 12s
15 actionable tasks: 7 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
-------
Test for io.opentelemetry:opentelemetry-exporter-common:1.19.0 passed.
====================

> Task :test

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 14s
11 actionable tasks: 3 executed, 8 up-to-date
PASSED:1.27.0
========================================================================================
 Testing io.opentelemetry:opentelemetry-exporter-common:1.19.0:1.28.0
========================================================================================
========================================================================================
 io.opentelemetry:opentelemetry-exporter-common:1.19.0:1.28.0 stage native-image run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 761ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest FAILED
====================
Testing library: io.opentelemetry:opentelemetry-exporter-common:1.19.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                               ^
  symbol:   class RetryPolicy
  location: package io.opentelemetry.exporter.internal.retry
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

                        () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                         ^
  symbol:   variable RetryPolicy
  location: class RetryUtilTest
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
  import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                                 ^
    symbol:   class RetryPolicy
    location: package io.opentelemetry.exporter.internal.retry
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                          () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                           ^
    symbol:   variable RetryPolicy
    location: class RetryUtilTest
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.


BUILD FAILED in 1s
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.opentelemetry:opentelemetry-exporter-common:1.19.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 1.28.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.28.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.28.0" ./gradlew clean test -Pcoordinates="io.opentelemetry:opentelemetry-exporter-common:1.19.0"]
========================================================================================
 io.opentelemetry:opentelemetry-exporter-common:1.19.0:1.28.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 745ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                               ^
  symbol:   class RetryPolicy
  location: package io.opentelemetry.exporter.internal.retry
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                        () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                         ^
  symbol:   variable RetryPolicy
  location: class RetryUtilTest
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
  import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                                 ^
    symbol:   class RetryPolicy
    location: package io.opentelemetry.exporter.internal.retry
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                          () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                           ^
    symbol:   variable RetryPolicy
    location: class RetryUtilTest
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 863ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'javac compile' for 1.28.0 failed with exit code 1.
FAILED[javac compile][1.28.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.28.0" ./gradlew clean compileTestJava -Pcoordinates="io.opentelemetry:opentelemetry-exporter-common:1.19.0"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.28.0" ./gradlew clean compileTestJava -Pcoordinates="io.opentelemetry:opentelemetry-exporter-common:1.19.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25027281349/job/73301240355
Last 300 lines of the log:
```console
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         2 tests found           ]
[         0 tests skipped         ]
[         2 tests started         ]
[         0 tests aborted         ]
[         2 tests successful      ]
[         0 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 18s
15 actionable tasks: 7 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
-------
Test for io.opentelemetry:opentelemetry-exporter-common:1.19.0 passed.
====================

> Task :test

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 20s
11 actionable tasks: 3 executed, 8 up-to-date
PASSED:1.27.0
========================================================================================
 Testing io.opentelemetry:opentelemetry-exporter-common:1.19.0:1.28.0
========================================================================================
========================================================================================
 io.opentelemetry:opentelemetry-exporter-common:1.19.0:1.28.0 stage native-image run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 763ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest FAILED
====================
Testing library: io.opentelemetry:opentelemetry-exporter-common:1.19.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                               ^
  symbol:   class RetryPolicy
  location: package io.opentelemetry.exporter.internal.retry
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                        () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                         ^
  symbol:   variable RetryPolicy
  location: class RetryUtilTest
2 errors


FAILURE: Build failed with an exception.

* What went wrong:
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
  import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                                 ^
    symbol:   class RetryPolicy
    location: package io.opentelemetry.exporter.internal.retry
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                          () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                           ^
    symbol:   variable RetryPolicy
    location: class RetryUtilTest
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.opentelemetry:opentelemetry-exporter-common:1.19.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 1.28.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.28.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.28.0" ./gradlew clean test -Pcoordinates="io.opentelemetry:opentelemetry-exporter-common:1.19.0"]
========================================================================================
 io.opentelemetry:opentelemetry-exporter-common:1.19.0:1.28.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 796ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                               ^
  symbol:   class RetryPolicy
  location: package io.opentelemetry.exporter.internal.retry
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date
                        () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                         ^
  symbol:   variable RetryPolicy
  location: class RetryUtilTest
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:13: error: cannot find symbol
  import io.opentelemetry.exporter.internal.retry.RetryPolicy;
                                                 ^
    symbol:   class RetryPolicy
    location: package io.opentelemetry.exporter.internal.retry
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.19.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java:38: error: cannot find symbol
                          () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
                                                                           ^
    symbol:   variable RetryPolicy
    location: class RetryUtilTest
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 869ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'javac compile' for 1.28.0 failed with exit code 1.
FAILED[javac compile][1.28.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.28.0" ./gradlew clean compileTestJava -Pcoordinates="io.opentelemetry:opentelemetry-exporter-common:1.19.0"]
```

--- a/metadata/io.opentelemetry/opentelemetry-exporter-common/1.28.0/reachability-metadata.json
+++ b/metadata/io.opentelemetry/opentelemetry-exporter-common/1.28.0/reachability-metadata.json
@@ -1,0 +1,10 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "io.opentelemetry.exporter.internal.auth.Authenticator"
+      },
+      "type": "io_opentelemetry.opentelemetry_exporter_common.AuthenticatorTest$DelegatingBuilder"
+    }
+  ]
+}

--- a/metadata/io.opentelemetry/opentelemetry-exporter-common/1.28.0/reachability-metadata.json
+++ b/metadata/io.opentelemetry/opentelemetry-exporter-common/1.28.0/reachability-metadata.json
@@ -1,10 +1,1 @@
-{
-  "reflection": [
-    {
-      "condition": {
-        "typeReached": "io.opentelemetry.exporter.internal.auth.Authenticator"
-      },
-      "type": "io_opentelemetry.opentelemetry_exporter_common.AuthenticatorTest$DelegatingBuilder"
-    }
-  ]
-}
+{ }

--- a/metadata/io.opentelemetry/opentelemetry-exporter-common/index.json
+++ b/metadata/io.opentelemetry/opentelemetry-exporter-common/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "1.28.0",
+    "source-code-url" : "https://repo1.maven.org/maven2/io/opentelemetry/opentelemetry-exporter-common/$version$/opentelemetry-exporter-common-$version$-sources.jar",
+    "repository-url" : "https://github.com/open-telemetry/opentelemetry-java",
+    "test-code-url" : "https://github.com/open-telemetry/opentelemetry-java/tree/v$version$/exporters/common/src/test",
+    "documentation-url" : "https://repo1.maven.org/maven2/io/opentelemetry/opentelemetry-exporter-common/$version$/opentelemetry-exporter-common-$version$-javadoc.jar",
+    "description" : "OpenTelemetry Exporter Common provides shared Java helper code used by OpenTelemetry exporters. It contains common internal exporter utilities for building telemetry export integrations without forcing optional transport dependencies onto every consumer.",
     "tested-versions" : [
       "1.28.0"
     ],

--- a/metadata/io.opentelemetry/opentelemetry-exporter-common/index.json
+++ b/metadata/io.opentelemetry/opentelemetry-exporter-common/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "1.28.0",
+    "tested-versions" : [
+      "1.28.0"
+    ],
+    "allowed-packages" : [
+      "io.opentelemetry.exporter.internal"
+    ]
+  },
+  {
     "metadata-version" : "1.19.0",
     "source-code-url" : "https://repo1.maven.org/maven2/io/opentelemetry/opentelemetry-exporter-common/$version$/opentelemetry-exporter-common-$version$-sources.jar",
     "repository-url" : "https://github.com/open-telemetry/opentelemetry-java",

--- a/stats/io.opentelemetry/opentelemetry-exporter-common/1.28.0/execution-metrics.json
+++ b/stats/io.opentelemetry/opentelemetry-exporter-common/1.28.0/execution-metrics.json
@@ -1,0 +1,101 @@
+{
+  "fix_javac_fail:2026-05-22": {
+    "timestamp": "2026-05-22T17:12:16.556810Z",
+    "library": "io.opentelemetry:opentelemetry-exporter-common:1.28.0",
+    "starting_commit": "90cf6d39e590be0c1aeb52e0f1d73b46dd228844",
+    "ending_commit": "f7f8031b87a62023c346e8e07c195882e4c97682",
+    "previous_library": "io.opentelemetry:opentelemetry-exporter-common:1.19.0",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "1.28.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 2,
+        "totalCalls": 2
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 143,
+          "missed": 5431,
+          "ratio": 0.025655,
+          "total": 5574
+        },
+        "line": {
+          "covered": 37,
+          "missed": 1306,
+          "ratio": 0.02755,
+          "total": 1343
+        },
+        "method": {
+          "covered": 8,
+          "missed": 312,
+          "ratio": 0.025,
+          "total": 320
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "1.19.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 0.5,
+            "coveredCalls": 2,
+            "totalCalls": 4
+          }
+        },
+        "coverageRatio": 0.5,
+        "coveredCalls": 2,
+        "totalCalls": 4
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 161,
+          "missed": 5675,
+          "ratio": 0.027587,
+          "total": 5836
+        },
+        "line": {
+          "covered": 40,
+          "missed": 1368,
+          "ratio": 0.028409,
+          "total": 1408
+        },
+        "method": {
+          "covered": 12,
+          "missed": 329,
+          "ratio": 0.035191,
+          "total": 341
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 42791,
+      "cached_input_tokens_used": 357376,
+      "output_tokens_used": 7018,
+      "iterations": 2,
+      "cost_usd": 0.3892,
+      "tested_library_loc": 37,
+      "metadata_entries": 0,
+      "test_only_metadata_entries": 3,
+      "previous_library_metadata_entries": 0,
+      "previous_library_test_only_metadata_entries": 2,
+      "code_coverage_percent": 2.76,
+      "previous_library_coverage_percent": 2.84
+    },
+    "artifacts": {
+      "test_file": "tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java",
+      "metadata_file": "metadata/io.opentelemetry/opentelemetry-exporter-common/1.28.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/io.opentelemetry/opentelemetry-exporter-common/1.28.0/stats.json
+++ b/stats/io.opentelemetry/opentelemetry-exporter-common/1.28.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "1.28.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 2,
+          "totalCalls" : 2
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 2,
+      "totalCalls" : 2
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 143,
+        "missed" : 5431,
+        "ratio" : 0.025655,
+        "total" : 5574
+      },
+      "line" : {
+        "covered" : 37,
+        "missed" : 1306,
+        "ratio" : 0.02755,
+        "total" : 1343
+      },
+      "method" : {
+        "covered" : 8,
+        "missed" : 312,
+        "ratio" : 0.025,
+        "total" : 320
+      }
+    }
+  } ]
+}

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/.gitignore
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/build.gradle
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "io.opentelemetry:opentelemetry-exporter-common:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/gradle.properties
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = io.opentelemetry:opentelemetry-exporter-common:1.28.0
+library.version = 1.28.0
+metadata.dir = io.opentelemetry/opentelemetry-exporter-common/1.28.0/

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/settings.gradle
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'io.opentelemetry.opentelemetry-exporter-common_tests'

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/AuthenticatorTest.java
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/AuthenticatorTest.java
@@ -1,0 +1,48 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_opentelemetry.opentelemetry_exporter_common;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.opentelemetry.exporter.internal.auth.Authenticator;
+import io.opentelemetry.exporter.internal.http.HttpExporterBuilder;
+import io.opentelemetry.exporter.internal.marshal.Marshaler;
+import java.util.Collections;
+import org.junit.jupiter.api.Test;
+
+public class AuthenticatorTest {
+    private final Authenticator authenticator = Collections::emptyMap;
+
+    @Test
+    void setsAuthenticatorOnHttpExporterDelegate() {
+        DelegatingBuilder builder =
+                new DelegatingBuilder(
+                        new HttpExporterBuilder<Marshaler>(
+                                "test-exporter", "telemetry", "http://localhost:4318/v1/test"));
+
+        assertThatCode(() -> Authenticator.setAuthenticatorOnDelegate(builder, authenticator))
+                .doesNotThrowAnyException();
+    }
+
+    @Test
+    void rejectsDelegateWithUnsupportedType() {
+        DelegatingBuilder builder = new DelegatingBuilder(new Object());
+
+        assertThatThrownBy(() -> Authenticator.setAuthenticatorOnDelegate(builder, authenticator))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Delegate field is not type");
+    }
+
+    private static final class DelegatingBuilder {
+        private final Object delegate;
+
+        private DelegatingBuilder(Object delegate) {
+            this.delegate = delegate;
+        }
+    }
+}

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java
@@ -1,0 +1,50 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_opentelemetry.opentelemetry_exporter_common;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.opentelemetry.exporter.internal.grpc.GrpcStatusUtil;
+import io.opentelemetry.exporter.internal.retry.RetryPolicy;
+import io.opentelemetry.exporter.internal.retry.RetryUtil;
+import org.junit.jupiter.api.Test;
+
+public class RetryUtilTest {
+    @Test
+    void exposesRetryableStatusCodes() {
+        assertThat(RetryUtil.retryableHttpResponseCodes())
+                .containsExactlyInAnyOrder(429, 502, 503, 504);
+        assertThat(RetryUtil.retryableGrpcStatusCodes())
+                .containsExactlyInAnyOrder(
+                        GrpcStatusUtil.GRPC_STATUS_CANCELLED,
+                        GrpcStatusUtil.GRPC_STATUS_DEADLINE_EXCEEDED,
+                        GrpcStatusUtil.GRPC_STATUS_RESOURCE_EXHAUSTED,
+                        GrpcStatusUtil.GRPC_STATUS_ABORTED,
+                        GrpcStatusUtil.GRPC_STATUS_OUT_OF_RANGE,
+                        GrpcStatusUtil.GRPC_STATUS_UNAVAILABLE,
+                        GrpcStatusUtil.GRPC_STATUS_DATA_LOSS);
+    }
+
+    @Test
+    void inspectsDelegateFieldWhenSettingRetryPolicy() {
+        DelegateHolder holder = new DelegateHolder(null);
+
+        assertThatThrownBy(
+                        () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("delegate field");
+    }
+
+    static final class DelegateHolder {
+        private final Object delegate;
+
+        private DelegateHolder(Object delegate) {
+            this.delegate = delegate;
+        }
+    }
+}

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/java/io_opentelemetry/opentelemetry_exporter_common/RetryUtilTest.java
@@ -7,10 +7,8 @@
 package io_opentelemetry.opentelemetry_exporter_common;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 import io.opentelemetry.exporter.internal.grpc.GrpcStatusUtil;
-import io.opentelemetry.exporter.internal.retry.RetryPolicy;
 import io.opentelemetry.exporter.internal.retry.RetryUtil;
 import org.junit.jupiter.api.Test;
 
@@ -31,20 +29,10 @@ public class RetryUtilTest {
     }
 
     @Test
-    void inspectsDelegateFieldWhenSettingRetryPolicy() {
-        DelegateHolder holder = new DelegateHolder(null);
-
-        assertThatThrownBy(
-                        () -> RetryUtil.setRetryPolicyOnDelegate(holder, RetryPolicy.getDefault()))
-                .isInstanceOf(IllegalArgumentException.class)
-                .hasMessageContaining("delegate field");
-    }
-
-    static final class DelegateHolder {
-        private final Object delegate;
-
-        private DelegateHolder(Object delegate) {
-            this.delegate = delegate;
-        }
+    void excludesNonRetryableStatusCodes() {
+        assertThat(RetryUtil.retryableHttpResponseCodes())
+                .doesNotContain(200, 400, 500);
+        assertThat(RetryUtil.retryableGrpcStatusCodes())
+                .doesNotContain("0", "3", "5");
     }
 }

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,12 @@
+{
+  "reflection": [
+    {
+      "fields": [
+        {
+          "name": "delegate"
+        }
+      ],
+      "type": "io_opentelemetry.opentelemetry_exporter_common.RetryUtilTest$DelegateHolder"
+    }
+  ]
+}

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,12 +1,18 @@
 {
-  "reflection": [
+  "reflection" : [
     {
-      "fields": [
+      "fields" : [
         {
-          "name": "delegate"
+          "name" : "delegate"
         }
       ],
-      "type": "io_opentelemetry.opentelemetry_exporter_common.RetryUtilTest$DelegateHolder"
+      "type" : "io_opentelemetry.opentelemetry_exporter_common.RetryUtilTest$DelegateHolder"
+    },
+    {
+      "condition" : {
+        "typeReached" : "io.opentelemetry.exporter.internal.auth.Authenticator"
+      },
+      "type" : "io_opentelemetry.opentelemetry_exporter_common.AuthenticatorTest$DelegatingBuilder"
     }
   ]
 }

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/user-code-filter.json
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "io.opentelemetry.exporter.internal.**"
+    }
+  ]
+}

--- a/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/user-code-filter.json
+++ b/tests/src/io.opentelemetry/opentelemetry-exporter-common/1.28.0/user-code-filter.json
@@ -5,6 +5,9 @@
     },
     {
       "includeClasses" : "io.opentelemetry.exporter.internal.**"
+    },
+    {
+      "includeClasses" : "io_opentelemetry.opentelemetry_exporter_common.**"
     }
   ]
 }

